### ISSUE TITLE
deno bundle vague error for bare imports

### ISSUE THREAD

**Issue by nayeemrmn** [denoland/deno#9570]
```ts
// temp.ts
import "foo";
```
`deno bundle temp.ts`
```
Bundle file:///mnt/c/Users/Nayeem/projects/deno/temp.ts
Check file:///mnt/c/Users/Nayeem/projects/deno/temp.ts
error: Unable to output bundle during Graph::bundle().
```

**Comment by kitsonk** [comment]
Yeah, we should not "swallow" the error from the module slot in the graph, but instead expose that... I think it was overuse of my `.context()` at one point in time.

While this isn't a good first issue, it is a good "medium" skill issue if someone wanted to get more familiar with the internals of Deno.

**Comment by wperron** [comment]
I can look into this 👍

**Comment by bartlomieju** [comment]
> Yeah, we should not "swallow" the error from the module slot in the graph, but instead expose that... I think it was overuse of my `.context()` at one point in time.

@kitsonk this sounds fishy to me, overuse of `context()` should result in multiple lines underneath "Caused by" when printing the error (for reference from `anyhow` crate: https://docs.rs/anyhow/1.0.32/anyhow/). If those lines are not visible it suggests something's wrong with our error formatting for terminal errors.

**Comment by bartlomieju** [comment]
Actually found a TODO, for exactly this problem:

https://github.com/denoland/deno/blob/687ff2ab14d6a735aa5bf0ec57ef00cfe0c04e4b/cli/main.rs#L1267-L1272

**Comment by wperron** [comment]
yeah I talked with @lucacasonato about it -- switching line 1271 for `format!("{:#}", e` does print out the error context so that's good, but the `relative import path "foo" not prefixed with...` message is being lost/dropped somewhere

**Comment by mitch292** [comment]
Hi - was browsing the l-rust tag and took a quick look at this one. Looks like the error is being swallowed on line 389:

https://github.com/denoland/deno/blob/bd961c3bc305e90e760394d2f424c5360a67543b/cli/module_graph.rs#L379-L395

Wasn't sure on the use case of when you wouldn't want to return the `Error` for `ModuleResolutionError::ImportPrefixMissing`, so didn't make a PR, but thought it might help.

**Comment by wperron** [comment]
@mitch292 yeah I saw that as well, but the same code works fine with `deno run` so I'm not sure if it's really the issue. In any case, if you have something working feel free to open a PR, I've been working on some other stuff lately and haven't had the chance to circle back to this issue

**Comment by mitch292** [comment]
@wperron Ahhhh I see. From issue #6618, it looks like the specific error that drives the `relative import path "foo" prefixed with..` message is only thrown at runtime to handle dynamic imports. I made a change per @bartlomieju comment to print the anyhow error chain, which at least provides a little more context. Hopefully this works, but if not, np. 

```
error: Unable to output bundle during Graph::bundle().

Caused by:
    0: load_transformed failed
    1: failed to analyze module
    2: failed to resolve foo from <file:///Users/andrewmitchell/Code/denocode/temp/temp.ts>
    3: The graph is missing a dependency.
         Specifier: foo from file:///Users/andrewmitchell/Code/denocode/temp/temp.ts
```

**Comment by wperron** [comment]
Fixed in #9822
